src/main/rp/ReplayRegistry.ts

```diff
@@ -0,0 +1,126 @@
+import { AuthorizationRequest } from '../authorization-request';
+import { AuthorizationResponse } from '../authorization-response';
+import { RequestObjectPayload } from '../types';
+import { AuthorizationEvents } from '../types/Events';
+import {
+  AuthorizationRequestState,
+  AuthorizationRequestStateStatus,
+  AuthorizationResponseState,
+  AuthorizationResponseStateStatus,
+} from '../types/ReplayRegistry';
+
+import { rpEventEmitter } from './RP';
+
+export class ReplayRegistry {
+  private authorizationRequests: Map<string, AuthorizationRequestState> = new Map<string, AuthorizationRequestState>();
+  private authorizationResponses: Map<string, AuthorizationResponseState> = new Map<string, AuthorizationResponseState>();
+
+  public constructor() {
+    rpEventEmitter.on(AuthorizationEvents.ON_AUTH_REQUEST_CREATED_SUCCESS, this.onAuthorizationRequestCreatedSuccess.bind(this));
+    rpEventEmitter.on(AuthorizationEvents.ON_AUTH_RESPONSE_RECEIVED_SUCCESS, this.onAuthorizationResponseReceivedSuccess.bind(this));
+    rpEventEmitter.on(AuthorizationEvents.ON_AUTH_RESPONSE_VERIFIED_SUCCESS, this.onAuthorizationResponseVerifiedSuccess.bind(this));
+    rpEventEmitter.on(AuthorizationEvents.ON_AUTH_RESPONSE_VERIFIED_FAILED, this.onAuthorizationResponseVerifiedFailed.bind(this));
+  }
+
+  private async onAuthorizationRequestCreatedSuccess(authorizationRequest: AuthorizationRequest): Promise<void> {
+    try {
+      if (!authorizationRequest) {
+        return;
+      }
+
+      const timestamp = Date.now();
+      const payload: RequestObjectPayload | undefined = await authorizationRequest.requestObject.getPayload();
+      if (!payload) {
+        throw new Error('Request does not contain a payload');
+      }
+      this.authorizationRequests.set(payload.nonce, { payload, status: AuthorizationRequestStateStatus.CREATED, timestamp, lastUpdated: timestamp });
+    } catch (error: any) {
```

Comment:
The any is generating warnings

Reply:
made it an unknown, as it is not an Error